lucene/luke/src/java/org/apache/lucene/luke/app/desktop/LukeMain.java

```diff
@@ -72,9 +72,14 @@ private static boolean createGUI() {
   public static void main(String[] args) throws Exception {
     boolean sanityCheck = Arrays.asList(args).contains("--sanity-check");
 
-    if (sanityCheck && GraphicsEnvironment.isHeadless()) {
-      Logger.getGlobal().log(Level.SEVERE, "[Vader] Hello, Luke. Can't do much in headless mode.");
-      Runtime.getRuntime().exit(0);
+    if (sanityCheck) {
+      if (GraphicsEnvironment.isHeadless()) {
+        Logger.getGlobal()
+            .log(Level.SEVERE, "[R2-D2] Hello, Luke. Can't do much in headless mode.");
```

Comment:
a trivial tip: I confirmed this line is skipped when the test runs on GH Actions; modifying this line is a convenient way to re-run all workflows without changing the execution path.